service/resource/endpoint/current.go

```diff
@@ -0,0 +1,38 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/canceledcontext"
+)
+
+const (
+	ipAnnotation      = "endpoint.kvm.giantswarm.io/ip"
+	serviceAnnotation = "endpoint.kvm.giantswarm.io/service"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+
+	pod, err := toPod(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	annotations, err := getAnnotations(*pod, []string{ipAnnotation, serviceAnnotation})
+	if err != nil {
+		canceledcontext.SetCanceled(ctx)
+		if canceledcontext.IsCanceled(ctx) {
+			r.logger.Log("pod", pod.GetName(), "debug", "canceling reconciliation for pod")
+			return nil, nil
+		}
+	}
+
+	endpoint := &Endpoint{
+		IP:               annotations[ipAnnotation],
+		ServiceName:      annotations[serviceAnnotation],
```

Comment:
Here again we blindly trust another implementation. I would rather suggest to let `getAnnotations` return exactly what we want. When I screw up it's implementation tomorrow I could just make this code panic.

Reply:
This seems pretty defensive, as this is a string map right now. But I changed the function to take two inputs and return two inputs, so now it returns exactly what we want.